src/java/com/puppetlabs/jruby_utils/pool/JRubyPool.java

```diff
@@ -192,15 +204,11 @@ public E borrowItemWithTimeout(long timeout, TimeUnit unit) throws
 
     @Override
     public void releaseItem(E e) {
-        releaseItem(e, true);
-    }
-
-    @Override
-    public void releaseItem(E e, boolean returnToPool) {
         final ReentrantLock lock = this.queueLock;
         lock.lock();
         try {
-            if (returnToPool) {
+            // Do nothing if the item being returned is the poisonPill
```

Comment:
> Future Us won't have to look at this and wonder "what happens if it's a different pill?"
Or even a different instance. Not sure how that'd happen in practice but I've read the code thought about that in the past. 

That seems like a good approach

>  I think having .insertPill calls beyond the first essentially be a no-op would be okay and maybe give us some flexibility for later.
Agreed with this as well 👍